File: src/main/java/com/tambapps/http/restclient/RestClient.java
package com.tambapps.http.restclient;

import com.tambapps.http.restclient.request.RestRequest;
import com.tambapps.http.restclient.response.handler.ResponseHandler;
import com.tambapps.http.restclient.response.handler.ResponseHandlers;
import com.tambapps.http.restclient.response.RestResponse;

/**
 * Class that sends synchronous REST requests
 */
public class RestClient extends AbstractRestClient {

  /**
   *
   * @param baseUrl the base url of the rest api
   */
  public RestClient(String baseUrl) {
    super(baseUrl);
  }

  public RestResponse<Void> execute(RestRequest request) {
    return execute(request, ResponseHandlers.noResponse());
  }

  public <T> RestResponse<T> execute(RestRequest request, ResponseHandler<T> responseHandler) {
    return doExecute(request, responseHandler);
  }

}

File: httpclient-okhttp3/src/main/kotlin/com/tambapps/http/client/RestClient.kt
package com.tambapps.http.client

/**
 * Class that sends synchronous Http requests to a rest service
 * every request will be prefixed by the base url, meaning that
 * Requests only contain the endpoint of the REST service
 */
class RestClient
/**
 *
 * @param baseUrl the base url of the rest api
 */
(baseUrl: String) : AbstractOkHttpHttpClient() {

    // ends without '/'
    private val baseUrl: String = formatBaseUrl(baseUrl)

    override fun getUrl(endpoint: String): String {
        return getRestUrl(baseUrl, endpoint)
    }
}